core/src/test/java/org/apache/calcite/util/FormatModelUtilTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.util;
+
+import org.apache.calcite.sql.FormatModel;
+import org.apache.calcite.util.format.FormatModelElement;
+import org.apache.calcite.util.format.FormatModelUtil;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Unit test for {@link FormatModelUtil}.
+ */
+public class FormatModelUtilTest {
+
+  private static final ImmutableMap TEST_MAP = FormatModel.BIG_QUERY_FORMAT_ELEMENT_PARSE_MAP;
+
+  private void assertFormatElementParse(String fmtStr, List<String> expected) {
```

Comment:
switch the 'expected' argument to a matcher, like this:
```
private void assertThatFormatElementParse(String fmtStr, Matcher<List<String>> matcher) {
```
so that you can write tests so that they can be read left-to-right in English:
```
assertThatFormatElementParse("%j", is(Arrays.asList("DDD")));
```